ocaml/xapi/xapi_xenops.ml

```diff
@@ -1132,6 +1132,20 @@ module MD = struct
       | `destroy ->
           [Vm.Shutdown]
     in
+    let on_softreboot_behaviour = function
```

Comment:
code here could be shared with on_crash: implement specific new behaviour for soft_reboot first and then hand off to on_crash